Update wazuh.rst to better explain how to use TLS (#1105)
stackhpc/stackhpc-kayobe-config: doc/source/configuration/wazuh.rst

   ``etc/kayobe/inventory/group_vars/wazuh-manager/wazuh-manager``, in
   particular the defaults assume that the ``provision_oc_net`` network will be
   used.

> #. Ensure to export vault password: ``export KAYOBE_VAULT_PASSWORD={ansible_vault_password}``

Reviewer: ```suggestion
#. Export the Ansible vault password e.g. : ``export KAYOBE_VAULT_PASSWORD=$(cat path/to/vault/password/file)``
```

Running this playbook will generate and put pertinent security items into secrets
vault file which will be placed in ``$KAYOBE_CONFIG_PATH/wazuh-secrets.yml``.
If using environments it ends up in ``$KAYOBE_CONFIG_PATH/environments/<env_name>/wazuh-secrets.yml``

> The secrets will be encrypted after templating and so requires that ``KAYOBE_VAULT_PASSWORD`` is set and exported before running the playbook!

Reviewer: ```suggestion
The secrets will be encrypted after templating. Ensure that ``KAYOBE_VAULT_PASSWORD`` is set and exported before running the playbook!
```

Wazuh secrets template is located in ``etc/kayobe/ansible/templates/wazuh-secrets.yml.j2``.
It will be used by wazuh secrets playbook to generate wazuh secrets vault file, which will then be encrypted.

.. code-block:: console

  kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/wazuh-secrets.yml

> To view ``wazuh-secrets.yml`` simply ``ansible-vault view --vault-password-file ~/vault.pass $KAYOBE_CONFIG_PATH/wazuh-secrets.yml``

Reviewer: ```suggestion
To view ``wazuh-secrets.yml``:

  ..code-block:: console

    ansible-vault view --vault-password-file path/to/vault/password/file $KAYOBE_CONFIG_PATH/wazuh-secrets.yml

```

Custom certificates for Wazuh ``indexer`` may be also be added in a similar way as Wazuh ``dashboard`` by changing the ``indexer_node_name`` such that it's going to match the custom certificate's name.
Once the certificates have been generated a number of steps are required, depending on which set of custom certificates are required to be added or substituted. Regardless, you must ``SSH`` into the ``infra VM`` in which the Wazuh services have been deployed to and, with admin privileges, remove any of the certificates from ``/etc/wazuh-{dashboard | indexer}/certs/`` which are due to be replaced with custom ones. Following this, return to the ``seed`` VM and drop the custom new ``pem`` certificates into ``$KAYOBE_CONFIG_PATH/environments/<environment>/wazuh/wazuh-certificates/`` if

> using a kayobe environment, or ``$KAYOBE_CONFIG_PATH/wazuh/wazuh-certificates/`` if not. Finally, rerun the Wazuh Ansible playbook and now the custom certificates should be copied over to their respective directories in the ``infra VM``.

Reviewer 2: Could you simplify some of the wording here? Phrases like "in order to generate the aforementioned certificates" are hard enough for my poor brain to parse. I can only imagine it'll be even harder for someone who's first language isn't English :)

Author: Should be a lot simpler to read now, let me know how you find it!

Reviewer: Much better now, but still some points need clearing up. See new review comments

> Following this, the ``sudo systemctl status`` for the three Wazuh services should be checked to make sure they are up and running with no errors. Additionally, testing ``sudo filebeat test output`` and trying to ``curl`` the service's IP will also signify if the certificates are working correctly.

Reviewer 2: Could these manual steps be turned into a playbook? The checks afterwards would be nice to have automated too

Author: I'm sure they could, but I believe the idea is to containerise Wazuh. Thus, I am not sure how that would affect this or if there is a point at the current moment.

Reviewer 2: Ah fair enough, let's leave it as-is and we can follow-up with automation if the containerisation never comes about.

- ``{{ dashboard_node_name }}-key.pem`` for the private key
- ``{{ dashboard_node_name }}.pem`` for the certificate

> In order to use externally generated certificates, you must first deploy Wazuh Ansible as normal in order to generate the base certificates. 
> This is because Wazuh will require to deploy with some sort of certificates, as well as, not every Wazuh service needing a custom external certificate; 
> for example the ``root`` and ``admin`` certificates. Therefore the ``<...>/wazuh/wazuh-certificates`` directory must be created and populated by ``wazuh-cert-tool.sh``, which it won't do if the ``<...>/wazuh/wazuh-certificates`` directory already exists.

Reviewer: While the existing text is correct, the "why" section makes it much more confusing. 

```suggestion
To use externally generated certificates, you must first deploy Wazuh Ansible as normal to generate the base certificates. The custom certificates can then be added to this bundle.
```

> Once the Wazuh generated certificates have been made, the desired custom external certificates need to be added in. 
> To do so, must ``SSH`` into the ``infra VM`` which Wazuh services have been deployed to and, with ``sudo`` privileges, 
> remove any of the 'base' certificates from ``/etc/wazuh-{dashboard | indexer}/certs/`` which are being replaced with custom ones. 
> After this, return to the ``seed`` VM and drop the custom new ``pem`` certificates into ``$KAYOBE_CONFIG_PATH/environments/<environment>/wazuh/wazuh-certificates/`` if using a kayobe environment, or ``$KAYOBE_CONFIG_PATH/wazuh/wazuh-certificates/`` if not. 
> Finally, rerun the Wazuh Ansible playbook and now the custom certificates should be copied over to their respective directories in the ``infra VM``.

Reviewer: Some changes here to wording, just to make it a bit clearer and more generic (e.g. not all wazuh-managers are infra VMs)

```suggestion
Once the Wazuh generated certificates have been made, the custom external certificates need to be added in. To do so, ssh into the Wazuh Manager and remove any of the 'base' certificates from ``/etc/wazuh-{dashboard | indexer}/certs/`` which are being replaced with custom ones.

Return to the ``seed`` and drop the custom ``pem`` certificates into ``$KAYOBE_CONFIG_PATH/environments/<environment>/wazuh/wazuh-certificates/`` if using a kayobe environment, or ``$KAYOBE_CONFIG_PATH/wazuh/wazuh-certificates/`` if not.

Finally, rerun the Wazuh Ansible playbook to deploy the custom certificates.
```

> Following this, the ``sudo systemctl status`` for the three Wazuh services should be checked to make sure they are up and running with no errors. Additionally, testing ``sudo filebeat test output`` and trying to ``curl`` the service's IP will also signify if the certificates are working correctly.

Reviewer: ```suggestion
Ensure the systemd units for the three Wazuh services are healthy. Additionally, testing ``sudo filebeat test output`` and trying to ``curl`` the service's IP will also signify if the certificates are working correctly.
```

PRIVATE KEY instead of BEGIN RSA PRIVATE KEY or BEGIN OPENSSH PRIVATE KEY).

Example OpenSSL rune to convert to PKCS#8:

``openssl pkcs8 -topk8 -nocrypt -in wazuh.key -out wazuh.key.pkcs8``

.. note::

    If you find that your Wazuh playbook isn't generating some of the non custom 'base' certificates,
    such as the ``indexer`` or ``dashboard`` certificates, it is likely that those services aren't being 
    templated correctly into the ``wazuh-cert-tool.sh`` which then creates the certificates. 
    Therefore it is likely that a variable in ``$KAYOBE_CONFIG_PATH/inventory/group_vars/wazuh-manager/wazuh-manager`` 
    is not matching the corresponding variable used to template ``wazuh-cert-tool.sh``.  

    Below is the template for generating ``wazuh-cert-tool.sh``; make sure all the ``"{{ variables }}"`` have been defined:

    .. code-block:: bash

        instances:
          wazuh:
            name: "{{ inventory_hostname }}"
            ip: "{{ wazuh_manager_ip }}"
            role: indexer

    Where ``role`` defines the service which ``wazuh-cert-tool.sh`` will create a certificate for.

>     This should be defined in ``$KAYOBE_CONFIG_PATH/environments/<environment>/inventory/group_vars/wazuh-manager/wazuh-manager.yml``.

Reviewer: I can't work out what this is saying
